synapseclient/core/async_utils.py

```diff
@@ -71,6 +73,28 @@ def f(*args, **kwds):
         return f
 
 
+def wrap_async_to_sync(coroutine, syn: "Synapse"):
```

Comment:
How difficult is it to type the coroutine?

Reply:
Not difficult - I'll add a Type hint:
`: Coroutine[Any, Any, Any]`

Since these are generic functions that take in any coroutine the type hints are less useful, but at least typed to a `Coroutine`